mail/api_test.py

```diff
@@ -576,3 +589,146 @@ def test_send_to_course_team_error(self, raise_for_status, mock_post):
+
+    def test_send_automatic_emails(self):
+        """send_automatic_emails should send emails to users which fit criteria and mark them so we don't send twice"""
+        with patch(
+            'mail.api.search_percolate_queries', autospec=True, return_value=[self.percolate_query],
+        ) as mock_search_queries, patch('mail.api.MailgunClient') as mock_mailgun:
+            send_automatic_emails(self.program_enrollment_unsent)
+
+        mock_search_queries.assert_called_with(self.program_enrollment_unsent.id)
+        mock_mailgun.send_individual_email.assert_called_with(
+            self.automatic_email.email_subject,
+            self.automatic_email.email_body,
+            self.program_enrollment_unsent.user.email,
+            sender_name=self.automatic_email.sender_name,
+        )
+
+    def test_no_matching_query(self):
+        """If there are no queries matching percolate we should do nothing"""
+        with patch(
+            'mail.api.search_percolate_queries', autospec=True, return_value=[],
+        ) as mock_search_queries, patch('mail.api.MailgunClient') as mock_mailgun:
+            send_automatic_emails(self.program_enrollment_unsent)
+
+        mock_search_queries.assert_called_with(self.program_enrollment_unsent.id)
+        assert mock_mailgun.send_individual_email.called is False
+
+    def test_not_enabled(self):
+        """If the automatic email is not enabled we should do nothing"""
+        with patch(
+            'mail.api.search_percolate_queries', autospec=True, return_value=[self.percolate_query_disabled],
+        ) as mock_search_queries, patch('mail.api.MailgunClient') as mock_mailgun:
+            send_automatic_emails(self.program_enrollment_unsent)
+
+        mock_search_queries.assert_called_with(self.program_enrollment_unsent.id)
+        assert mock_mailgun.send_individual_email.called is False
+
+    def test_already_sent(self):
+        """If a user was already sent email we should not send it again"""
+        with patch(
+            'mail.api.search_percolate_queries', autospec=True, return_value=[self.percolate_query],
+        ) as mock_search_queries, patch('mail.api.MailgunClient') as mock_mailgun:
+            send_automatic_emails(self.program_enrollment_sent)
+
+        mock_search_queries.assert_called_with(self.program_enrollment_sent.id)
+        assert mock_mailgun.send_individual_email.called is False
+
+    def test_failed_send(self):
+        """If we fail to send email to the first user we should still send it to the second"""
+
+        with mute_signals(post_save):
+            new_percolate = PercolateQuery.objects.create(
+                query="new query",
+                original_query="original query",
+            )
+        new_automatic = AutomaticEmail.objects.create(
+            enabled=True,
+            email_subject='new subject',
+            email_body='new body',
+            sender_name='new Sender name',
+            query=new_percolate,
+        )
```

Comment:
Once factories are created, please replace these too. Might just be handy to do a full codebase search for `AutomaticEmail.objects.create` and `PercolateQuery.objects.create`